### PR TITLE
Kamu UI 299 add support env vars and secrets for dataset flows from UI

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**
Command for launch:
docker run -it --rm -p 8080:8080 -e KAMU_AUTH_GITHUB_CLIENT_ID=361a3b4fda86d0234d2f -e KAMU_AUTH_GITHUB_CLIENT_SECRET=465849325236ed49253993744069e1bec6808554 -e DATASET_ENV_VAR_ENCRYPTION_KEY=QfnEDcnUtGSW2pwVXaFPvZOwxyFm2BOC [ghcr.io/kamu-data/kamu-base:latest-with-dataset-env-var-storage](http://ghcr.io/kamu-data/kamu-base:latest-with-dataset-env-var-storage) kamu system api-server --address 0.0.0.0 --http-port 8080 --external-address 127.0.0.1

Result:

image

image

image

image

image

image

### CODE REVIEW

File: src/app/api/environment-variables.api.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                {
+                    update: (cache) => {
+                        const datasetKeyFragment = DatasetApi.generateDatasetKeyFragment(
+                            cache.identify(DatasetApi.generateAccountKeyFragment(params.accountId)),
+                            params.datasetId,
+                        );
+                        cache.evict({
+                            id: cache.identify(datasetKeyFragment),
+                            fieldName: "envVars",
+                        });
```
Comment (zaychenko-sergei) [inline src/app/api/environment-variables.api.ts:66]: We should have some helper class to reset dataset object parts. This is complex enough to copy paste.

File: src/app/api/environment-variables.api.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        return this.modifyEnvVariableGQL.mutate({ ...params }).pipe(
+            first(),
+            map((result: MutationResult<ModifyEnvVariableMutation>) => {
+                /* istanbul ignore else */
+                if (result.data) {
+                    return result.data;
+                } else {
+                    throw new DatasetOperationError(result.errors ?? []);
+                }
+            }),
```
Comment (zaychenko-sergei) [inline src/app/api/environment-variables.api.ts:98]: Shouldn't modify reset cache too?

File: src/app/api/environment-variables.api.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        return this.deleteEnvVariableGQL.mutate({ ...params }).pipe(
+            first(),
+            map((result: MutationResult<DeleteEnvVariableMutation>) => {
+                /* istanbul ignore else */
+                if (result.data) {
+                    return result.data;
+                } else {
+                    throw new DatasetOperationError(result.errors ?? []);
+                }
+            }),
```
Comment (zaychenko-sergei) [inline src/app/api/environment-variables.api.ts:115]: What about delete - what's the effect from cache point of view?

File: src/app/services/evnironment-variables.service.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    ModifyEnvVariableMutation,
+    SaveEnvVariableMutation,
+    ViewDatasetEnvVarConnection,
+} from "../api/kamu.graphql.interface";
+import { ToastrService } from "ngx-toastr";
+
+@Injectable({
+    providedIn: "root",
+})
+export class EvnironmentVariablesService {
```
Comment (zaychenko-sergei) [inline src/app/services/evnironment-variables.service.ts:17]: This feature is specific to datasets, but you've named it as if we were talking about some global env vars. Use word "dataset" in the service name or in methods names.

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]
image

A lot of coverage still missing for the new components.

File: src/app/api/environment-variables.api.ts (hunk 1 above)
Comment (dmitriy-borzenko) [reply src/app/api/environment-variables.api.ts:66]: Moved to helper method

File: src/app/api/environment-variables.api.ts (hunk 2 above)
Comment (dmitriy-borzenko) [reply src/app/api/environment-variables.api.ts:98]: Yes, added

File: src/app/services/evnironment-variables.service.ts (hunk 4 above)
Comment (dmitriy-borzenko) [reply src/app/services/evnironment-variables.service.ts:17]: Renamed

File: src/app/api/environment-variables.api.ts (hunk 3 above)
Comment (dmitriy-borzenko) [reply src/app/api/environment-variables.api.ts]: Fixed

File: src/app/common/data.helpers.ts [hunk 5]
```diff
@@ -304,3 +305,24 @@ export const noCacheFetchPolicy: {
...
+        cache.identify(DatasetApi.generateAccountKeyFragment(params.accountId)),
+        params.datasetId,
+    );
+    params.fieldNames.forEach((fieldName: string) => {
+        cache.evict({
+            id: cache.identify(datasetKeyFragment),
+            fieldName,
+        });
+    });
+};
```
Comment (zaychenko-sergei) [inline src/app/common/data.helpers.ts:328]: We have `apollo-cache.helper.ts` for cache stuff

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**